UNITTESTS/fakes/events/EventQueue.h

```diff
@@ -0,0 +1,90 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef EVENTQUEUE_FAKE_H
+#define EVENTQUEUE_FAKE_H
+
+#include <map>
+#include <memory>
+#include <functional>
+#include <algorithm>
+#include "events/EventQueue.h"
+#include <chrono>
+
+namespace events {
+
+typedef intptr_t handle_t;
+typedef std::function<void()> function_t;
+typedef unsigned long tick_t;
+
+class EventQueue {
+    using duration = std::chrono::duration<int, std::milli>;
+
+public:
+    EventQueue(unsigned size = 0, unsigned char *buffer = NULL) { delete buffer; };
+
+    ~EventQueue() { };
+
+    void dispatch(int ms = -1) {
+        if (ms > 0) {
+            process_events(ms);
+        } else {
+            dispatch_forever();
+        }
+    };
+
+    void dispatch_forever() {
+        now = (tick_t)-1;
+        process_events();
+        now = 0;
+    };
+
+    void break_dispatch() { };
+
+    tick_t tick() { return now; };
+
+    bool cancel(handle_t id) { return cancel_handler(id); };
+
+    handle_t call(function_t f)
```

Comment:
The real API is inconvenient here, it returns an hardcoded `int` and this returns an intptr. I would like this function to follow the prototype and return an int. 
As a solution, a counter can be used and incremented everytime an event is added. In another map the association between the ID and the associated element in `handlers` is maintained. 
Elements are removed in both map when an event is processed or cancelled. 

I also think parameters can follow the original API: 

```c++
template<typename F, typename ... Args>
void call(F f, Args... args) { 
    _call([f, args = mstd::make_tuple(args...)] () { 
        mstd::apply(f, args);
    });
}


void _call(std::function<void()> f);

```

`mstd::apply` and make_tupe should be available in the [mstd_tuple](https://github.com/ARMmbed/mbed-os/blob/master/platform/cxxsupport/mstd_tuple). 

With this technique it should be easy to add the pointer to member overload.

Reply:
thanks, added method versions and multiple argument support

Reply:
and call functions now return int as the handle